src/test/java/jenkins/bouncycastle/EncodignDecodingTest.java

```diff
@@ -224,5 +226,15 @@ public void testCreationFromObjectPrivateKeyPEM() throws Exception {
 
         assertEquals(FileUtils.readFileToString(PRIVATE_KEY_PEM), FileUtils.readFileToString(pemFileNew));
     }
+    
+    
+    @Test
+    @Issue(value="JENKINS-35661") 
+    public void testReadKeyPairFromPCKS8PEM() throws Exception {
+        PEMEncodable pemEnc = PEMEncodable.read(getResourceFile("private-key-pcks8.pem"));
+        assertNotNull(pemEnc.toKeyPair());
+        assertNotNull(pemEnc.toPrivateKey());
+        assertNotNull(pemEnc.toPublicKey());
```

Comment:
check that the PEM encoding round-trip gives the same form and that the public key matches. I suspect that this code may be assuming the public exponent is 65537 so RSA keys using a different public exponent may fail for this code (not that there should be any, but theoretically....)